### ISSUE TITLE
Unable to update Departments unless newly created

### ISSUE THREAD

**Issue by sylphaxiom** [snipe/snipe-it#7083]
#### Please confirm you have done the following before posting your bug report:

- [ x ] I have enabled debug mode 
- [ x ] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

**Describe the bug**
When I try to edit previously added departments, I get a pop up stating " Error: Please check the form below for errors" When trying to cancel, the page refreshes with no changes. Same thing happens when clicking the "Back" button.

**Expected behavior**
The Department should update after putting in the information.

**Screenshots**
![image](https://user-images.githubusercontent.com/35042829/58505924-1a50fb80-815c-11e9-9209-8f3deb8d1921.png)

**Server (please complete the following information):**
 - Snipe-IT Version v4.6.16 - build 4018
 - OS: Ubuntu
 - Web Server: Apache 2
 - PHP Version: 7.2.17

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser Chrome
 - Version 74

**Error Messages**
No debug errors on Laravel panel or browser developer tools.

**Additional context**
* I've never manually entered data into the database.
* Manual install with no errors or issues
* No changes since the last time this was working.
* Still able to edit Departments that were added today, but not ones added days ago.
* No errors in any server logs.

Please do not post an issue without answering the related questions above. If you have opened a different issue and already answered these questions, answer them again, once for every ticket. It will be next to impossible for us to help you.

**Comment by snipe** [comment]
In the debugbar, please click on the Requests tab, and tell me what is in the expanded errors->message bag section in session_attributes, for example:

<img width="937" alt="Screen Shot 2019-05-28 at 12 28 55 PM" src="https://user-images.githubusercontent.com/197404/58506493-58422580-8144-11e9-873e-add09112d660.png">

**Comment by sylphaxiom** [comment]
Session Attributes:
![image](https://user-images.githubusercontent.com/35042829/58506799-48cfd600-815e-11e9-8a9b-b523e63f0cad.png)
Response Headers:
![image](https://user-images.githubusercontent.com/35042829/58506840-5d13d300-815e-11e9-8336-8c69de705c03.png)
Request Cookies:
![image](https://user-images.githubusercontent.com/35042829/58506874-7026a300-815e-11e9-92b4-d5159ee342f4.png)
Request Server:
![image](https://user-images.githubusercontent.com/35042829/58506915-816faf80-815e-11e9-97ef-2097cd325544.png)
Request Headers:
![image](https://user-images.githubusercontent.com/35042829/58506957-9c422400-815e-11e9-8d8e-ad8e23d2a938.png)
All:
![image](https://user-images.githubusercontent.com/35042829/58507025-c3005a80-815e-11e9-9360-e436c8005736.png)

**Comment by sylphaxiom** [comment]
I do get an error in the Sessions tab as follows:
![image](https://user-images.githubusercontent.com/35042829/58507174-2094a700-815f-11e9-8772-27dbe24e2c3f.png)

**Comment by sylphaxiom** [comment]
It does not appear to happen with ALL previously created Departments but just previously created departments that were created with only a Name field filled in. If I attempt to create a new one with only a name field, and go back to edit it, I have no issues.

**Comment by snipe** [comment]
That's useful information, thank you. Is it possible the original user who created the ones that fail was deleted at some point?

**Comment by sylphaxiom** [comment]
Shouldn't be. I'm the one implementing this system for the city and I'm the only user who has inputted any information or done anything with it. Anyone else with permissions to log in has only viewed the site briefly. I haven't trained them on it yet. And I haven't modified my own account. I also tried to log out and log back in and still have the same issue.

**Comment by snipe** [comment]
The constraint it's catching on based on that error message is:

`'user_id'               => 'nullable|exists:users,id',`

though. That's the only check it does for a user in that process, so my guess is that department was created by someone other than you, or was imported so the user ID is coming up as invalid.

In `app/Models/Departments.php`, remove the line mentioned above, which should be around line 29. So this:

```
 protected $rules = [
        'name'                  => 'required|max:255',
        'user_id'               => 'nullable|exists:users,id',
        'location_id'           => 'numeric|nullable',
        'company_id'            => 'numeric|nullable',
        'manager_id'            => 'numeric|nullable',
    ];
```

should look like this:

```
 protected $rules = [
        'name'                  => 'required|max:255',
        'location_id'           => 'numeric|nullable',
        'company_id'            => 'numeric|nullable',
        'manager_id'            => 'numeric|nullable',
    ];
```

and tell me if that works.

**Comment by sylphaxiom** [comment]
Yes, that did fix the issue.

Is this a permanent fix? Or would this create a security hole?

**Comment by sylphaxiom** [comment]
It appears the user_id has been removed from the rows that are having this issue. Not sure why.
![image](https://user-images.githubusercontent.com/35042829/58508164-6f434080-8161-11e9-9910-b7db77a1f7c2.png)

**Comment by snipe** [comment]
I don't know why those would be `0`, but user_id isn't really used anywhere important for that table, more just general knowledge of who created it.

I wouldn't suggest doing something, even temporarily, that would create a security hole. I'm pushing this change out into the next release. There may be times when a user_id isn't entered correctly for a department, and since it's not important for business logic, we should just bypass that check.

Likewise, if a user creates a department, they get deleted and purged (meaning soft-deletes are actually removed), no one will ever be able to edit that department again, which is bad.

**Comment by sylphaxiom** [comment]
Sounds good. I'll keep that line out and call it good! Thank you very much!

**Comment by snipe** [comment]
No problem :) FYI, this ticket will be automatically closed once the 4.7 branch containing this fix is merged into master (which should be today)
